core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -141,31 +141,39 @@ public void registerWithDeprecatedHandler(RestRequest.Method method, String path
     }
 
     /**
-     * Registers a REST handler to be executed when the provided method and path match the request.
+     * Registers a REST handler to be executed when one of the provided methods and path match the request.
      *
-     * @param method GET, POST, etc.
      * @param path Path to handle (e.g., "/{index}/{type}/_bulk")
      * @param handler The handler to actually execute
+     * @param methods GET, POST, etc.
      */
-    public void registerHandler(RestRequest.Method method, String path, RestHandler handler) {
-        PathTrie<RestHandler> handlers = getHandlersForMethod(method);
-        if (handlers != null) {
-            handlers.insert(path, handler);
-            if (handler instanceof BaseRestHandler) {
-                usageService.addRestHandler((BaseRestHandler) handler);
-            }
-        } else {
-            throw new IllegalArgumentException("Can't handle [" + method + "] for path [" + path + "]");
+    public void registerHandler(String path, RestHandler handler, RestRequest.Method... methods) {
+        if (handler instanceof BaseRestHandler) {
+            usageService.addRestHandler((BaseRestHandler) handler);
         }
+        handlers.insertOrUpdate(path, new MethodHandlers(path, handler, methods), (mHandlers, newMHandler) -> {
+            return mHandlers.addMethods(methods, handler);
+        });
+    }
+
+    /**
+     * Registers a REST handler to be executed when the provided method and path match the request.
+     *
+     * @param singleMethod GET, POST, etc.
+     * @param path Path to handle (e.g., "/{index}/{type}/_bulk")
+     * @param handler The handler to actually execute
+     */
+    public void registerHandler(RestRequest.Method singleMethod, String path, RestHandler handler) {
+        RestRequest.Method[] methods = new RestRequest.Method[1];
```

Comment:
why do you construct this array rather thank just passing it on to the var args? and why do we have this extra method wouldn't it be ok to just have the one with the var args?